moduleframework/tools/modulelint.py

```diff
@@ -71,25 +71,96 @@ def test_run_or_usage_label_exists(self):
         self.assertTrue(label_found)
 
 
+class DockerfileLinterInContainer(container_avocado_test.ContainerAvocadoTest):
+    """
+    :avocado: enable
+
+    """
+
+    def _file_to_check(self, doc_file_list):
+        test_failed = False
+        for doc in doc_file_list.split('\n'):
+            exit_status = self.run("test -e %s" % doc, ignore_status=True).exit_status
+            if int(exit_status) == 0:
+                self.log.debug("%s doc file exists in container" % doc)
+                test_failed = True
+        return test_failed
+
+    def test_all_nodocs(self):
+        """
+        This test checks all docs installed in container. As delivered by Platform
+        as delivered by installing packages RUN command
+        :return: WARNING in case we found some docs
+                 PASS in case there is no doc file found
+        """
+        self.start()
+        all_docs = self.run("rpm -qad", verbose=False).stdout
+        test_failed = self._file_to_check(all_docs)
+        if test_failed:
+            self.log.warn("Documentation files exist in container. They are installed by Platform or by RUN commands.")
+        self.assertTrue(True)
+
+    def test_installed_docs(self):
+        """
+        This test checks whether no docs are installed by RUN dnf command
+        :return: FAILED in case we found some docs
+                 PASS in case there is no doc file found
+        """
+        self.start()
+        installed_pkgs = self.run("rpm -qa --qf '%{{NAME}}\n'", verbose=False).stdout
+        defined_pkgs = self.backend.getPackageList()
+        list_pkg = set(installed_pkgs).intersection(set(defined_pkgs))
+        test_failed = False
+        for pkg in list_pkg:
+            pkg_doc = self.run("rpm -qd %s" % pkg, verbose=False).stdout
+            if self._file_to_check(pkg_doc):
+                test_failed = True
+        self.assertFalse(test_failed)
+
+    def test_docker_clean_all(self):
+        """
+        This test checks if size of /var/cache/<pkg_manager> is bigger then
```

Comment:
this function is still wrong in my opinion

Reply:
What do you prefer? I would like to have something which will work for this time.
Parsing image layers one by one will follow soon.

@ignatenkobrain What is your preference? Please paste code.

Reply:
@ignatenkobrain I am still waiting for your proposal. I will merge it on Friday, nevertheless.

Reply:
I told this many times. Do `clean all` unconditionally when building images.

Reply:
@ignatenkobrain the response from the team who is working on the build system was negative - they advised not to do such magic, hence we need to figure out the linter